lib/cache_manager.rb

```diff
@@ -14,7 +15,12 @@ def initialize(cache)
   def read(name, _options = nil)
     filtered_name = name.to_s.parameterize(separator: '')
     if File.exist?(File.join(@cache, filtered_name))
-      File.read(File.join(@cache, filtered_name), mode: 'r')
+      content = File.read(File.join(@cache, filtered_name), mode: 'r')
+      begin
+        Zlib::Inflate.inflate(content)
+      rescue Zlib::DataError # Previously dumped files were not compressed, protect them
```

Comment:
is it possible to check instead file type ? 
- txt --> no need inflate
- other --> inflate

Reply:
`Zlib::Inflate.inflate` has an efficient internal check in fact.

It checks the header flags (https://stackoverflow.com/a/30794147/1200528) and returns immediately without doing much if it doesn't see the magic headers. Since we already have the content, I think this is less error prone compared to maintaining a flag or checking file type manually. What do you say Fabien?